src/storage/src/source/types.rs

```diff
@@ -182,10 +173,18 @@ pub enum NextMessage<Key, Value, Diff> {
 pub enum SourceMessageType<Key, Value, Diff> {
     /// Communicate that this [`SourceMessage`] is the final
     /// message its its offset.
-    Finalized(SourceMessage<Key, Value, Diff>),
+    Finalized(
+        Result<SourceMessage<Key, Value>, SourceReaderError>,
+        (PartitionId, MzOffset),
+        Diff,
```

Comment:
I suggest making a struct -- or at least a type alias -- for this triple, since it's used ~everywhere.

Reply:
Unfortunately it's not really worth doing this because very timely and DD need to have these three types as separate type parameters so whatever I do to group them I'll have to undo in the very next PR that's coming that makes these actual timely streams

Reply:
Okay! (Normally I'd expect a type alias to be fine either way but I will reserve judgement to the next PR.)